common/util.go

```diff
@@ -83,18 +85,18 @@ func HandleHTTPRequest(message string, headers http.Header, data ConnectorMetada
 		}
 		if err == nil && resp.StatusCode >= 200 && resp.StatusCode < 300 {
 			// Success, quit retrying
-			return resp, nil
+			return &resp.StatusCode, resp, nil
 		}
 	}
 
 	if resp == nil {
-		return nil, fmt.Errorf("every function invocation retry failed; final retry gave empty response. http_endpoint: %v, source: %v", data.HTTPEndpoint, data.SourceName)
+		return nil, nil, fmt.Errorf("every function invocation retry failed; final retry gave empty response. http_endpoint: %v, source: %v", data.HTTPEndpoint, data.SourceName)
 	}
 
 	if resp.StatusCode < 200 || resp.StatusCode > 300 {
-		return nil, fmt.Errorf("request returned failure: %v. http_endpoint: %v, source: %v", resp.StatusCode, data.HTTPEndpoint, data.SourceName)
+		return &resp.StatusCode, resp, fmt.Errorf("request returned failure: %v. http_endpoint: %v, source: %v", resp.StatusCode, data.HTTPEndpoint, data.SourceName)
 	}
-	return resp, nil
+	return &resp.StatusCode, resp, nil
```

Comment:
I would like to understand the reason why we are returning status code if it can be extracted from response.

Reply:
I was thinking we should have status code, response body(instead of whole response) or error . If we look at how connectors are using it, it made sense to me if we do it that way. What are your thoughts?

Reply:
I agree we should utilize status code, but the same status code can be read from the response that way it's more pragmatic and avoid duplication. Think of this package as an HTTP package which only returns response and error.

Reply:
It wouldn't be duplicate if we have status code , body and error as paramaters, right?

Reply:
The core reason for this change was to allow an error handler to read the response headers, not just the body in the case of an error. This is to support headers for kafka - so that we can control partitioning and compaction.

We have 3 states:
* Request could not. be sent (network down etc)
* Request sent, but responded with an error - in this case with the current setup we get a nil for response. 
* Success. 

If we move to 'body' only, we'll lose a chance to handle partitioning and compaction or any of the more advanced features of kafka (at least for errors).

Reply:
@ranoble So do you suggest we should read the status code from the response object itself?

Reply:
That was the initial proposal. The concern was that you'd have both a response AND an error. 
I don't really see this as a problem (but then I'm not a golang native, so I'd follow your guidance), as you can safely ignore the response if you don't need it, but use it if you do (in the kafka case).

Reply:
It is also backwards compatible - and I always prefer that...

Reply:
@ranoble From the code changes you made, it seems you have introduced following changes:
1. Returning status code from function
2.  At line no. 97 you are returning resp instead of nil `if resp.StatusCode < 200 || resp.StatusCode > 300`
I am good with the second change as we would have more info about error which can be utilized while writing to error topic/queue. I don't think first change is necessary as we can derive this from response directly.

Reply:
@ranoble From the code changes you made, it seems you have introduced following changes:

Returning status code from function
At line no. 97 you are returning resp instead of nil if resp.StatusCode < 200 || resp.StatusCode > 300
Second change makes sense as we would have more info about error which can be utilized while writing to error topic/queue. I don't think first change is necessary as we can get status code from response directly.